Improve SANO sample (#507)
temporalio/samples-go: nexus-standalone-operations/README.md

This sample demonstrates how to use Standalone Nexus Operations (executing Nexus operations directly from client code without wrapping them in a Workflow).
It shows both sync and async (workflow-backed) operations, and how to use the `ListNexusOperations` and `CountNexusOperations` APIs.

> The starter and worker connect to two different namespaces (a "caller" namespace and a "target" namespace) — this mirrors how Nexus is typically used to cross namespace boundaries. The client is configured via the SDK's [environment configuration](https://docs.temporal.io/develop/environment-configuration) support, which reads `TEMPORAL_NAMESPACE`, `TEMPORAL_ADDRESS`, etc. from the environment (and optionally profiles from `temporal.toml`).

Reviewer: ```suggestion
The starter and worker connect to two different namespaces (a "caller" namespace and a "handler" namespace) — this mirrors how Nexus is typically used to cross namespace boundaries. The client is configured via the SDK's [environment configuration](https://docs.temporal.io/develop/environment-configuration) support, which reads `TEMPORAL_NAMESPACE`, `TEMPORAL_ADDRESS`, etc. from the environment (and optionally profiles from `temporal.toml`).
```

## Note: Standalone Nexus operations require a server version that supports this feature. Use the dev server build at https://github.com/temporalio/cli/releases/tag/v1.7.3-standalone-nexus-operations.

## Run locally against a dev server

> 1) Start the [Temporal dev server build that supports standalone Nexus operations](https://github.com/temporalio/cli/releases/tag/v1.7.3-standalone-nexus-operations) with the required namespaces pre-created:

Reviewer: ```suggestion
1) Start the [Temporal dev server build that supports standalone Nexus operations](https://docs.temporal.io/standalone-nexus-operation#temporal-cli-support) with the required namespaces pre-created:
```

Start the dev server with the dynamic config flags required for standalone Nexus operations:
1) Start the [Temporal dev server build that supports standalone Nexus operations](https://github.com/temporalio/cli/releases/tag/v1.7.3-standalone-nexus-operations) with the required dynamic config and both namespaces pre-created:

```bash
temporal server start-dev \

>   --dynamic-config-value "nexusoperation.enableStandalone=true" \

Author: This will require a new CLI release, just to note

Reviewer 2: Was this added in [v1.7.4-standalone-nexus-operations](https://github.com/temporalio/cli/releases/tag/v1.7.4-standalone-nexus-operations)?

Author: Yep!

>   --namespace nexus-standalone-caller-namespace \

Reviewer: ```suggestion
  --namespace my-caller-namespace \
```

>   --namespace nexus-standalone-target-namespace

Reviewer: ```suggestion
  --namespace my-handler-namespace
```

> 2) Create a Nexus endpoint that routes to the target namespace and the worker's task queue:

Reviewer: ```suggestion
2) Create a Nexus endpoint that routes to the handler namespace and the worker's task queue:
```

```bash
./temporal operator nexus endpoint create \

>   --name hello-service \

Reviewer: ```suggestion
  --name my-nexus-endpoint \
```

>   --target-namespace nexus-standalone-target-namespace \

Reviewer: ```suggestion
  --target-namespace my-handler-namespace \
```

>   --target-task-queue nexus-standalone-operations

Reviewer: ```suggestion
  --target-task-queue nexus-handler-queue
```

```

> 3) In a new terminal, start the worker in the target namespace:

Reviewer: ```suggestion
3) In a new terminal, start the worker in the handler namespace:
```

```bash

> TEMPORAL_NAMESPACE=nexus-standalone-target-namespace \

Reviewer: ```suggestion
TEMPORAL_NAMESPACE=my-handler-namespace \
```

  go run nexus-standalone-operations/worker/main.go
```

You should see a log line similar to:

```bash

> 2026/05/21 08:59:49 INFO  Started Worker Namespace nexus-standalone-target-namespace TaskQueue nexus-standalone-operations WorkerID 71172

Reviewer: ```suggestion
2026/05/21 08:59:49 INFO  Started Worker Namespace my-handler-namespace TaskQueue nexus-handler-queue WorkerID 71172
```

```

4) In a third terminal, run the starter in the caller namespace:

```bash

> TEMPORAL_NAMESPACE=nexus-standalone-caller-namespace \

Reviewer: ```suggestion
TEMPORAL_NAMESPACE=my-caller-namespace \
```

2026/05/21 09:00:30 Total Nexus operations: 2
```

If you run the starter multiple times, additional entries will appear in the `ListNexusOperations` output and the `CountNexusOperations` total will grow.

## Run against Temporal Cloud

> 1) Create two namespaces in Temporal Cloud (for example `my-caller-namespace.<account>` and `my-target-namespace.<account>`) and generate an API key (or mTLS cert) that can access both.

Reviewer: ```suggestion
1) Create two namespaces in Temporal Cloud (for example `my-caller-namespace.<account>` and `my-handler-namespace.<account>`) and generate an API key (or mTLS cert) that can access both.
```

> 2) Create a Nexus endpoint that routes to the target namespace and the worker's task queue. See the Temporal Cloud instructions at https://docs.temporal.io/nexus/registry#create-a-nexus-endpoint. Use:

Reviewer: ```suggestion
2) Create a Nexus endpoint that targets the handler namespace and the worker's task queue. See the Temporal Cloud instructions at https://docs.temporal.io/nexus/registry#create-a-nexus-endpoint. Use:
```

> - Endpoint name: `hello-service`

Reviewer: ```suggestion
- Endpoint name: `my-nexus-endpoint`
```

> - Target namespace: `my-target-namespace.<account>`

Reviewer: ```suggestion
- Target namespace: `my-handler-namespace.<account>`
```

> - Target task queue: `nexus-standalone-operations`

Reviewer: ```suggestion
- Target task queue: `nexus-handler-queue`
```

- Allowed caller namespaces: include `my-caller-namespace.<account>` (endpoints reject callers that are not on this list)

3) Add two profiles to your [environment configuration file](https://docs.temporal.io/develop/environment-configuration), one per namespace. Using API keys:

```toml

> [profile.target]

Reviewer: ```suggestion
[profile.handler]
```

address = "<region>.<cloud>.api.temporal.io:7233"

> namespace = "my-target-namespace.<account>"

Reviewer: ```suggestion
namespace = "my-handler-namespace.<account>"
```

api_key = "<your-api-key>"

[profile.caller]
address = "<region>.<cloud>.api.temporal.io:7233"
namespace = "my-caller-namespace.<account>"
api_key = "<your-api-key>"
```

For mTLS instead of API keys, set `tls.client_cert_path` and `tls.client_key_path` on each profile (see the [docs](https://docs.temporal.io/develop/environment-configuration) for the full schema).

4) Run the worker and starter in separate terminals, selecting the appropriate profile in each:

```bash

> # terminal 1 (worker, target namespace)

Reviewer: ```suggestion
# terminal 1 (worker, handler namespace)
```

> export TEMPORAL_PROFILE="target"

Reviewer: ```suggestion
export TEMPORAL_PROFILE="handler"
```
